### PR TITLE
Versioning Proposal

### PR DESCRIPTION
**Pull request by mfine**
Here's a first pass through versioning policy and generated artifacts in major version directories. Comments very welcome.

+ Updated the versioning policy to cover message stability.
+ Updated the generators for c, python, latex - this is kind of a mess and might need a refactor.
+ C headers generated; non-header files left as is (and common.h - not using a relative path!)
+ Python generated. Didn't update the client code here or the tables stuff.

Mainly looking to see what we think about this approach.

/cc @fnoble @mookerji @cbeighley

### CODE REVIEW

File: VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+In order to support backwards and forwards compatibility, SBP messages
+are versioned and follow a small number of conventions, enabling
+devices to indicate understood message formats, capabilities and
+features. SBP uses a major, minor numbering scheme to indicate
+versions, where major and minor are represented by 8-bit values. The
+minor number is incremented by each message addition, deprecation. The
+major number is incremented when the format of a message is
```
Comment (mookerji) [inline VERSIONING.md]: The clarifyL "message" here refers to the SBP packet structure or the change of a stable message?

Comment (mfine) [inline VERSIONING.md]: Change of a stable message. There's no provision here for changing the SBP framing structure. Should we include some language do disallow that?

File: VERSIONING.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+changed.
+
+Message ID's are guaranteed to be unique for all minor versions in a
```
Comment (mookerji) [inline VERSIONING.md]: An alternative layout would make this easier to parse? Possibly something like ... 

* *Major version*. The major number is incremented when the format of a message is changed. Message ID's are not guaranteed to be unique between major versions. The addition or deprecation of stable messages results in the incrementing of the major version. Stable messages are guaranteed to be available for all minor versions in a major version

* *Minor version*.  The minor number is incremented by each non-stable message addition, deprecation.  Changes to a message in a major version are manifest by incrementing the minor version, adding a new message with the changes, and deprecating the old message.

File: c/include/libsbp/v0/acquisition.h [hunk 3]
```diff
@@ -20,10 +20,10 @@
  * * Satellite acquisition messages from the device.
  * \{ */
 
-#ifndef LIBSBP_ACQUISITION_MESSAGES_H
-#define LIBSBP_ACQUISITION_MESSAGES_H
+#ifndef LIBSBP_V0_ACQUISITION_MESSAGES_H
+#define LIBSBP_V0_ACQUISITION_MESSAGES_H
 
-#include "common.h"
+#include "<libsbp/common.h>"
```
Comment (mookerji) [inline c/include/libsbp/v0/acquisition.h]: 0_0 mixed quotation angle brackets?

File: generator/sbpg/targets/resources/sbp_messages_template.h [hunk 4]
```diff
@@ -20,10 +20,10 @@
  * (((description|commentify)))
  * \{ */
 
-#ifndef LIBSBP_(((pkg_name|upper)))_MESSAGES_H
-#define LIBSBP_(((pkg_name|upper)))_MESSAGES_H
+#ifndef LIBSBP_V(((major)))_(((pkg_name|upper)))_MESSAGES_H
+#define LIBSBP_V(((major)))_(((pkg_name|upper)))_MESSAGES_H
 
-#include "common.h"
+#include "<libsbp/common.h>"
```
Comment (mookerji) [inline generator/sbpg/targets/resources/sbp_messages_template.h]: ditto

File: VERSIONING.md (hunk 1 above)
Comment (mookerji) [inline VERSIONING.md]: Just wanted to make sure that's what it was referring to. 

A mention of stable vs unstable/draft, similar to what we have in the datasheet would be good: "By convention, we refer to messages as `stable` and `draft`. Stable messages are intended for broad use, like `MsgBaselineECEF`, while draft ones are intended for internal development.`.

File: generator/sbpg/targets/resources/sbp_messages_template.h [hunk 5]
```diff
@@ -53,4 +53,4 @@ typedef struct __attribute__((packed)) {
 ((* endfor *))
 /** \} */
 
-#endif /* LIBSBP_(((pkg_name|upper)))_MESSAGES_H */
+#endif /* LIBSBP_V(((major)))_(((pkg_name|upper)))_MESSAGES_H */
```
Comment (fnoble) [inline generator/sbpg/targets/resources/sbp_messages_template.h]: Could we add a newline at the end of the file?

File: VERSIONING.md (hunk 2 above)
Comment (fnoble) [inline VERSIONING.md]: Agree, I think the content is spot on but could be restructured to be easier to read.

**Comment by mfine** [comment]
Completion criteria: completed `versioning.md` and version containers.

**Pull request closed without merging**